Classes/EventStore/EventStore.php

```diff
@@ -52,6 +55,22 @@ public function get(EventStreamFilterInterface $filter): EventStream
         return $eventStream;
     }
 
+    /**
+     * Returns the first EventAndRawEvent instance that matches the given $filter - or throws an EventNotFoundException if no matching events could be found
+     *
+     * @param EventStreamFilterInterface $filter
+     * @return EventAndRawEvent
+     * @throws EventNotFoundException
+     */
+    public function getOne(EventStreamFilterInterface $filter): EventAndRawEvent
+    {
+        $rawEvent = $this->storage->loadOne($filter);
+        if ($rawEvent === null) {
+            throw new EventNotFoundException('No event found for the given filter', 1513170518);
+        }
+        return (new EventStream(new \ArrayIterator([$rawEvent])))->current();
```

Comment:
Ugh...

Reply:
True, that's not the most beautiful piece of code and we should centralize event (de)serialization. But without that in place we'd have to duplicate a lot of code & dependencies